src/sxml-generic_serialize.ads

```diff
@@ -23,7 +23,8 @@ is
                         Data     : out String;
                         Last     : out Natural;
                         Result   : out Result_Type) with
-     Pre => Data'Length > 0;
+     Global => (In_Out => State),
+     Pre    => Data'Length > 0;
```

Comment:
This procedure should probably also have `Post => (if Result = Result_OK then Last in Data'Range)`.

Reply:
Indeed. Except that `Last` - contrary to what the name implies - is not the index of the last element but the offset to the last element. I changed the name to `Offset` and added a postcondition that `Offset < Data'Range` in 92a3669. `Parse` got an equivalent post condition (independent of the parse result, as the offset to the error location is given in the case of an error).